1-Draft/RFCNNNN-VSCode-PSScriptAnalyzer-Rules.md

```diff
@@ -0,0 +1,151 @@
+| AvoidUsingComputerNameHardcoded              | Error       | Default    |                |
+| AvoidUsingConvertToSecureStringWithPlainText | Error       | Default    |                |
+| AvoidUsingDeprecatedManifestFields           | Warning     | Aggressive | x              |
+| AvoidUsingEmptyCatchBlock                    | Warning     |            |                |
+| AvoidUsingInvokeExpression                   | Warning     |            |                |
+| AvoidUsingPlainTextForPassword               | Warning     | Default    |                |
+| AvoidUsingPositionalParameters               | Warning     | Aggressive |                |
+| AvoidTrailingWhitespace                      | Warning     |            |                |
+| AvoidUsingUsernameAndPasswordParams          | Error       | Aggressive |                |
+| AvoidUsingWMICmdlet                          | Warning     | Aggressive | x              |
+| AvoidUsingWriteHost                          | Warning     | Aggressive |                |
+| DSCDscExamplesPresent                        | Information |            |                |
+| DSCDscTestsPresent                           | Information |            |                |
+| DSCReturnCorrectTypesForDSCFunctions         | Information |            |                |
+| DSCStandardDSCFunctionsInResource            | Error       |            |                |
+| DSCUseIdenticalMandatoryParametersForDSC     | Error       |            |                |
+| DSCUseIdenticalParametersForDSC              | Error       |            |                |
+| DSCUseVerboseMessageInDSCResource            | Error       |            |                |
+| MisleadingBacktick                           | Warning     | Default    |                |
+| MissingModuleManifestField                   | Warning     | Default    |                |
+| PossibleIncorrectComparisonWithNull          | Warning     | Default    |                |
+| PossibleIncorrectUsageOfAssignmentOperator   | Warning     | Aggressive |                |
+| PossibleIncorrectUsageOfRedirectionOperator  | Warning     | Default    |                |
+| ProvideCommentHelp                           | Information |            |                |
+| ReservedCmdletChar                           | Error       | Default    |                |
+| ReservedParams                               | Error       | Default    |                |
+| ShouldProcess                                | Error       | Default    |                |
+| UseApprovedVerbs                             | Warning     | Default    |                |
+| UseBOMForUnicodeEncodedFile                  | Warning     |            |                |
+| UseCmdletCorrectly                           | Warning     | Aggressive |                |
+| UseDeclaredVarsMoreThanAssignments           | Warning     | Default    |                |
```

Comment:
I would not enable this by default but in the more thorough mode. Or alternatively not use it in `.tests.ps1` files as PSSA is not Pester aware and the rule is limited to a per scriptblock basis.